modules/fuzzy/include/opencv2/fuzzy/fuzzy_F1_math.hpp

```diff
@@ -0,0 +1,152 @@
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#ifndef __OPENCV_FUZZY_F1_MATH_H__
+#define __OPENCV_FUZZY_F1_MATH_H__
+
+#include "opencv2/fuzzy/types.hpp"
+#include "opencv2/core.hpp"
+
+namespace cv
+{
+
+namespace ft
+{
+    //! @addtogroup f1_math
+    //! @{
+
+    /** @brief Computes components of the array using direct F1-transform.
+    @param matrix Input array.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param components Output 32-bit array for the components.
+
+    The function computes linear components using predefined kernel.
+
+    @note
+        F-transform technique of first degreee is described in paper @cite Vlas:FT.
+    */
+    CV_EXPORTS_W void FT12D_components(InputArray matrix, InputArray kernel, OutputArray components);
+
+    /** @brief Computes elements of F1-transform components.
+    @param matrix Input array.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param c00 Elements represent average color.
+    @param c10 Elements represent average vertical gradient.
+    @param c01 Elements represent average horizontal gradient.
+    @param components Output 32-bit array for the components.
+    @param mask Mask can be used for unwanted area marking.
+
+    The function computes components and its elements using predefined kernel and mask.
+
+    @note
+        F-transform technique of first degreee is described in paper @cite Vlas:FT.
+    */
+    CV_EXPORTS_AS(FT12D_polynomial1) void FT12D_polynomial(InputArray matrix, InputArray kernel, OutputArray c00, OutputArray c10, OutputArray c01, OutputArray components, InputArray mask);
+
+    /** @brief Computes elements of F1-transform components.
+    @param matrix Input array.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param c00 Elements represent average color.
+    @param c10 Elements represent average vertical gradient.
+    @param c01 Elements represent average horizontal gradient.
+    @param components Output 32-bit array for the components.
+
+    The function computes components and its elements using predefined kernel.
+
+    @note
+        F-transform technique of first degreee is described in paper @cite Vlas:FT.
+    */
+    CV_EXPORTS_W void FT12D_polynomial(InputArray matrix, InputArray kernel, OutputArray c00, OutputArray c10, OutputArray c01, OutputArray components);
```

Comment:
This is interesting. I did this because of problems with Python wrapper. Anyway, I tried now and everything works. I will fix it in F0 transform as well.